patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+## Title
+
+Extensions to Manage Contributions at Scale
+
+## Patlet
+
+TBD
+
+## Problem
+
+As the number of new feature contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping repositories well maintained?
+
+## Context
+
+- More employees are contributing to strategic InnerSource codebases, some of which are scaling rapidly with new feature contributions. This is adding a huge code review backlog on a smaller group of maintainers, resulting in maintainer burnout as well as slowing down release of new capabilities.
+- Adding an excessive number of capabilities and code to repositories is making them difficult to maintain.
+- here is a growing backlog of new features and ideas for the products which have scaled.
+- The pattern applies in either scenario:
+   - Maintainers find themselves rejecting new feature ideas to narrow down the scope of the products. This is hampering innovation in the community and restricting further expansion.
+   - To reduce backlog, new features are getting released without thorough documentation, hardening, or testing, creating a poor user experience. This is also bloating the size of the codebases, adding a huge dependency graph and making them difficult to maintain.
```

Comment:
- it mainly means a lot of code or utilities added to one repository
- many dependencies added, which would increase size of installed project
- more tests to maintain, and more code to maintain in general

Reply:
Is the codebase bloating happen due to "new features are getting released without thorough documentation, hardening, or testing".

Or would this bloating even happen if the maintainers had enough time to review and integrate the contributions thoroughly?

Reply:
For our specific application, which is a project where new capabilities are added on a regular basis to support more use-cases and things are rarely taken out, the "bloating" here is directly correlated with the number of things added to the primary repository.

Having more time to review and integrate the contributions could help for _some_ pieces of this. For example, for the growing stack of dependencies, in many cases you could take the time to rewrite a capability to avoid a new dependency. However, I'd say this bloat is to some degree unavoidable if most of the following are true:

- The use-case the primary repository tackles has a big feature space, and there is no plan for the rough number of capabilities in the project to hit steady-state, i.e., nontrivial things are often added and rarely deprecated/removed 
- The project is very important to many stakeholders, e.g., products, which accelerates deadlines. More user demands can make it hard to prioritize large tech debt to fix this.
- The project has a huge number of contributors, which is much larger than the maintainer group. This problem compounds onto the others, especially if contributors are trying to add new capabilities simultaneously

Reply:
hi @alex-jw-brooks . Great to see you here, and thank you for your work on this pattern.

What you are describing here could be added to the [Story](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/meta/pattern-template.md#story-optional) section of the pattern. Possibly even together with the maintainer burnout bits described [in here](https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/444#discussion_r961141465) as both of these seem specific to your project. 

I don't think that these aspects have been discussed in other patterns before, so sharing them could lead to interesting follow-up patterns and discussions.

Reply:
I was thinking about adding our story to that section as well, do you mind if we do it after merge of initial pattern though (after sept 5th)? we do have an interesting backstory to share I think and we may have more pattern ideas that we implemented to solve the burnout issues. Its essentially one story that could lead to a few patterns.

We will try to open up another issue as well with a community mentoring scaling model that we applied.
We also want to mention these issues as part of a case study we have coming up with InnerSourceCommons and IBM